### ISSUE TITLE
Act differently on build-number option of flutter build aar & flutter build apk (or aab)

### ISSUE THREAD

**Issue by wangmir** [flutter/flutter#98108]
Simply, I realized that the build-number option is acting weirdly.

for pubspec.yaml with

```yaml
version: 1.0.0
```

when providing --build-number option to 
`flutter build apk --build-number 3`,

build apk with version 1.0.0+3.

But when providing --build-number option to 
`flutter build aar --build-number 3`,

aar version is set with just "3", not 1.0.0+3. It seems not to be intented right..?

**Comment by maheshmnj** [comment]
Hi @wangmir, Have you tried adding version name in place of version code  `flutter build aar --build-number 1.0.0`?
as mentioned here https://github.com/flutter/flutter/issues/52731#issuecomment-625845663

**Comment by wangmir** [comment]
> Hi @wangmir, Have you tried adding version name in place of version code `flutter build aar --build-number 1.0.0`? as mentioned here [#52731 (comment)](https://github.com/flutter/flutter/issues/52731#issuecomment-625845663)

Yes, and that was worked, but I mean it seems not consistent with apk option. In the perspective of source version control,  is it right to follow `pubspec.yaml` rather than insert version number from --build-number?

And moreover, the help doc of `flutter build apk --help` and `flutter build aar --help` are exactly same. (while flutter build apk has another option --build-name which can give version number like 1.0.0 rather than build number)

```
    --build-number                  An identifier used as an internal version number.
                                    Each build must have a unique identifier to differentiate it from previous builds.
                                    It is used to determine whether one build is more recent than another, with higher numbers indicating more recent build.
                                    On Android it is used as "versionCode".
                                    On Xcode builds it is used as "CFBundleVersion".
```

**Comment by maheshmnj** [comment]
> I think if flutter build aar also follows the version number of pubspec.yaml first, it will be more convenient for maintaining source.

Thanks for the info, that seems like a plausible proposal. Labeling this issue for further insights from the team.

**Comment by wangmir** [comment]
@maheshmnj thanks 👍
